### PR TITLE
Local testing

### PR DESCRIPTION
**Pull request by chicco785**
Added functionalities to test locally the controller. Local test are more complete that the ones that can be achieved in Travis CI (unless we use three external VMs).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,12 +84,44 @@ Few hints, to customize the [`docker-compose.yml`](docker-compose.yml) orchestra
...
+### Prerequisites
+
+* A [Docker Swarm cluster](https://docs.docker.com/engine/swarm/swarm-tutorial/create-swarm/) (docker >= 17.06.0-ce) (locally or in the cloud as you prefer).
+* [shunit2](https://github.com/kward/shunit2) installed on your client
+* (Optionally) [VirtualBox](http://virtualbox.org) if you want to use the scripts provided in `utils` folder.
+* Your Docker client configured to point to the Docker Swarm cluster.
+
+### Utilities
+To test the script you need to set-up a Docker Swarm cluster. Assuming that VirtualBox is installed on your Linux/MacOS system you, can use the scripts in the `utils` folder to create and manage the cluster:
+* [create-cluster.sh](utils/create-cluster.sh) creates a Docker Swarm cluster with 1 Manager and 2 Workers (you can change the number of the Workers)
```
Comment (taliaga) [inline README.md:99]: I believe this repo is not the right place for these scripts.
For the recipes, I've been using https://github.com/aelsabbahy/miniswarm, which I have referenced in the "utils" of the recipes repo.

File: README.md [hunk 2]
```diff
@@ -84,12 +84,44 @@ Few hints, to customize the [`docker-compose.yml`](docker-compose.yml) orchestra
...
 
+## Testing
+
+### Prerequisites
+
+* A [Docker Swarm cluster](https://docs.docker.com/engine/swarm/swarm-tutorial/create-swarm/) (docker >= 17.06.0-ce) (locally or in the cloud as you prefer).
+* [shunit2](https://github.com/kward/shunit2) installed on your client
+* (Optionally) [VirtualBox](http://virtualbox.org) if you want to use the scripts provided in `utils` folder.
+* Your Docker client configured to point to the Docker Swarm cluster.
+
```
Comment (taliaga) [inline README.md]: Add [docker-machine](https://docs.docker.com/machine/install-machine/)

File: test/test-locally.sh [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+testSetUpMongoClusterSize(){
+  result=$(docker exec client mongo --quiet mongo_mongo/admin --eval "db.runCommand( { replSetGetStatus : 1 } )['members'].length")
+  assertEquals "3" "${result}"
+}
+
+# Kill container
+
+testKillMongo(){
```
Comment (taliaga) [inline test/test-locally.sh:46]: The are some like this one which are only testing docker behavior. I'd get rid of those.

All in all, is it me or there is some implicit "dependency" among these tests? If so, that'd not be good.

OK, re-reading everything, it looks to me some which are defined as tests are actually prerequisites for the actual tests located a bit below. If that's the case, I'd merge those together in a single test.

**Review by taliaga — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (chicco785) [reply README.md:99]: for the time being i would keep as it is. we have other priorities and this is enough for our needs. by using a different script, we will need either to change "fixed" names in the tests or to work out generic ways to retrieve the names of the docker machines to connect to. (which may not be easy if you have multiple docker-machine environments). Please add it as issue.

File: test/test-locally.sh (hunk 3 above)
Comment (chicco785) [reply test/test-locally.sh:46]: it is surely the case (and this is to save time in the testing to re-set every time the environment). given the priorities, for the time being i would leave this unchanged. please add it as issue.

Comment (taliaga) [reply test/test-locally.sh:46]: Ok, no prob. Considering your comment in time-saving, merging in a single test method would be worse then. If they start bothering we can transform those docker-only ones to "setup" methods to be called. No need of issue for now.

File: README.md [hunk 4]
```diff
@@ -84,12 +84,47 @@ Few hints, to customize the [`docker-compose.yml`](docker-compose.yml) orchestra
...
 - [x] The repository includes a basic set of Travis CI tests that tests the script behavior against basic conditions in a single swarm node and without data persistence (initialization, scale up, scale down)
 
+## Testing
+
+### Prerequisites
+
+* A [Docker Swarm cluster](https://docs.docker.com/engine/swarm/swarm-tutorial/create-swarm/) (docker >= 17.06.0-ce) (locally or in the cloud as you prefer).
+* [shunit2](https://github.com/kward/shunit2) installed on your client
+* (Optionally) [VirtualBox](http://virtualbox.org) if you want to use the scripts provided in `utils` folder.
+* (Optionally) [docker-machine](https://docs.docker.com/machine/install-machine/) if you use the scripts provided in `utils` folder.
```
Comment (taliaga) [inline README.md:94]: Actually it's not optional for testing, is it?

**Review by taliaga — APPROVED** [review]

**Pull request merged**